### PR TITLE
52069-show-course

### PR DESCRIPTION
**Pull request by l0ngvh**
## Related Tickets
- [#52069](https://edu-redmine.sun-asterisk.vn/issues/52069)

## WHAT (optional)
- I added a page to show all lesson in a course like a map path

## HOW
- I create a separate component for lessons nodes in the map, and combine them together using javascript.

## WHY (optional)
- It's more appealing this way

## Evidence (Screenshot or Video)
image
image
image

### CODE REVIEW

**Comment by htpt2910** [comment]
okeee

**Comment by l0ngvh** [comment]
ready

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -4,4 +4,8 @@ class CoursesController < BaseController
   def index
     @pagy, @courses = pagy Course.recommended
   end
+
+  def show
+    @course = Course.find_by id: params[:id]
```
Comment (vanvtt-0952) [inline app/controllers/courses_controller.rb:9]: thêm code handle khi course nil bé ơi

Comment (ngocvt-0484) [inline app/controllers/courses_controller.rb:9]: chưa control @course k tìm thấy

File: spec/components/previews/lesson_node_component_preview.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class LessonNodeComponentPreview < ViewComponent::Preview
+  layout "signin"
+  def with_completed
+    lesson = Lesson.new
+    lesson.completed = true
+    render(LessonNodeComponent.new(lesson: lesson))
+  end
+
+  def with_incompleted
+    lesson = Lesson.new
```
Comment (ngocvt-0484) [inline spec/components/previews/lesson_node_component_preview.rb]:
```suggestion
    lesson = Lesson.new completed: false
```

File: spec/components/previews/course_map_component_preview.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class CourseMapComponentPreview < ViewComponent::Preview
+  layout "signin"
+  def with_default
+    lessons = Lesson.first 13
```
Comment (ngocvt-0484) [inline spec/components/previews/course_map_component_preview.rb]: cho 13 vào setting

File: spec/components/previews/course_map_component_preview.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    render(CourseMapComponent.new(lessons: lessons))
+  end
+
+  def with_completed
+    lessons = Lesson.first 13
+
+    lessons = lessons.map do |lesson|
```
Comment (ngocvt-0484) [inline spec/components/previews/course_map_component_preview.rb]: thế dùng each là dc cần gì dùng map

**Comment by l0ngvh** [comment]
ready

**Pull request merged**